Sources/LocalReceiptParsing/BasicTypes/AppleReceipt.swift

```diff
@@ -60,6 +60,19 @@ struct AppleReceipt: Equatable {
 
 }
 
+// MARK: - Extensions
+
+extension AppleReceipt {
+
+    func containsActivePurchase(forProductIdentifier identifier: String) -> Bool {
+        return (self.inAppPurchases.contains { $0.isActiveSubscription } ||
+                self.inAppPurchases.contains { $0.productId == identifier })
```

Comment:
Since you're doing `||`... Wouldn't this statement return `true` if there's an active subscription for a product identifier? It would also return true if there's an inactive subscription for that productId.

Reply:
Yeah, I _think_ this is the logic we ended up agreeing on to avoid false negatives during a presubscription. @aboedo did I do this right?

Reply:
ohhh I see what @vegaro means, though - if you made _any_ purchases for a past subscription this will now be true. 
it should be either of: 
- you have any active subscriptions at all (this is just in case of upgrades, where product id won't be the same as the one you purchased)
- you have a non-subscription purchase for that product identifier
But if you only have an expired subscription for that product identifier this should be false

Reply:
Okay I'll update this logic 👍🏻 

Once that's finished, are we okay merging this PR enabled only for sandbox then? At the very least it will help integration tests. We could also only enable it for tests.

Reply:
we can start by having it for tests only and figure out next steps once we have a clearer picture of how to tackle triage-82

Reply:
That would be risk-free, since the only difference is a separate fetch policy that for now is unused :)

Reply:
Okay logic updated.